lib/spandex_ecto/telemetry_adapter.ex

```diff
@@ -0,0 +1,27 @@
+defmodule SpandexEcto.TelemetryAdapter do
+  @moduledoc """
+  This module provides event handler functions for telemetry
+  """
+
+  alias SpandexEcto.EctoLogger
+
+  #  this is for ecto_sql 3.0.x
+  def handle_event([_app_name, repo_name, :query], total_time, log_entry, _config) when is_integer(total_time) do
+    EctoLogger.trace(log_entry, "#{repo_name}_database")
+  end
+
+  # This is for ecto_sql >= 3.1
+  def handle_event([_app_name, repo_name, :query], measurements, metadata, _config) when is_map(measurements) do
+    log_entry = %{
+      query: metadata.query,
+      source: metadata.source,
+      params: metadata.params,
+      query_time: Map.get(measurements, :query_time, 0),
+      decode_time: Map.get(measurements, :decode_time, 0),
+      queue_time: Map.get(measurements, :queue_time, 0),
+      result: {metadata.result, "n/a"}
```

Comment:
I think they're passing the correct values again in `metadata.result` after this revert:
https://github.com/elixir-ecto/ecto_sql/commit/11b83c58b97c1d4a7f0a2a882f048929f294eee6

Reply:
Thanks again @novaugust. From the commit you mentioned, it looks like to me `metadata.result` used to be an `:ok` or `:error` atom, which is why you wrapped it in a tuple; but now it's a tuple already, so wrapping is not necessary anymore?

Reply:
Right. I was compensating for their breaking change in ecto 3.1.0 - they patched it in 3.1.1, see their changelog for the details =)

Reply:
@novaugust Thanks for the confirmation. I've added a new commit to handle both atom and tuple for metadata.result